awx/main/models/credential/__init__.py

```diff
@@ -275,6 +280,10 @@ class Meta:
         'admin_role',
     ])
 
+    def __init__(self, *args, **kwargs):
+        super(Credential, self).__init__(*args, **kwargs)
+        self.dynamic_input_fields = self._get_dynamic_input_field_names()
```

Comment:
This is going to be a real major performance issue, because any time a list of credentials is loaded, which would have otherwise been `O(1)` queries, this will incur `O(N)` queries.

Reply:
Yeah - Instead of doing this, I was thinking about doing the query the first time `get_input` is called and caching the field names.